tools/hookgen/HookGen.cpp

```diff
@@ -214,8 +218,8 @@ HookGen::writeEventToPublicHeader(const char *name, const char *description, con
 	char *exampleBuf = (char *)malloc(bufSize + 1);
 	if (NULL != exampleBuf) {
 		/* If we fail to allocate room for the example just skip writing it */
-		int length = sprintf(exampleBuf, (char *)example, name, description, name, structName);
-		exampleBuf[length] = '\0';
+		snprintf(exampleBuf, bufSize + 1, (char *)example, name, description, name, structName);
+		exampleBuf[bufSize] = '\0';
```

Comment:
I might be miscounting, but I think the three "%s" format tags in `example` will result in `bufSize` being three more than necessary, so I think `exampleBuf[bufSize] = '\0'` will place a null character too far after the end of the formatted output.  However, given that both `snprintf` and `sprintf` will null terminate `exampleBuf` already, I think this assignment can just go away, can't it?

Reply:
Miscounting, yes: there are four (on lines 1, 2, 4 and 9 of example). The NUL written here will be after the one written by `snprintf()`, but not "too far after" (it's still within the allocated buffer).

Reply:
Sorry for the confusion - by "too far after", I just meant after the point where it will be meaningful, not past the point of safety.

Reply:
I removed the `exampleBuf[bufSize] = '\0';` line.
Both `snprintf()` and `sprintf()` safely terminate the string within the allocated buffer in this case.